File: go/src/OpenAPIKeyClient.go
package main

import (
	"crypto/hmac"
	"crypto/sha256"
	"crypto/tls"
	"encoding/hex"
	"encoding/json"
	"fmt"
	"net"
	"net/url"
	"strconv"
	"strings"
	"time"

	"github.com/google/uuid"
	"github.com/valyala/fasthttp"
)

// FastHttpClientOption 定义HTTP客户端配置选项
type FastHttpClientOption struct {
	Header         map[string]string
	SocketTimeout  int // 毫秒
	ConnectTimeout int // 毫秒
	IgnoreSSL      bool
	ProxyAddress   string
}

// FastHttpClient HTTP客户端
type FastHttpClient struct {
	client *fasthttp.Client
	header map[string]string
}

// NewFastHttpClient 创建新的HTTP客户端
func NewFastHttpClient(option FastHttpClientOption) *FastHttpClient {
	client := &fasthttp.Client{
		ReadTimeout:  time.Duration(option.SocketTimeout) * time.Millisecond,
		WriteTimeout: time.Duration(option.SocketTimeout) * time.Millisecond,
		Dial: func(addr string) (net.Conn, error) {
			return fasthttp.DialTimeout(addr, time.Duration(option.ConnectTimeout)*time.Millisecond)
		},
		TLSConfig: &tls.Config{
			InsecureSkipVerify: option.IgnoreSSL,
		},
	}

	// 代理配置暂时注释掉，因为fasthttp.NewDialer在某些版本中可能不可用
	// if option.ProxyAddress != "" {
	// 	client.Dial = func(addr string) (net.Conn, error) {
	// 		return fasthttp.Dial(addr)
	// 	}
	// }

	return &FastHttpClient{
		client: client,
		header: option.Header,
	}
}

// Get 发送GET请求
func (c *FastHttpClient) Get(url string, headers map[string]string) (*fasthttp.Response, error) {
	req := fasthttp.AcquireRequest()
	defer fasthttp.ReleaseRequest(req)

	resp := fasthttp.AcquireResponse()
	defer fasthttp.ReleaseResponse(resp)

	req.SetRequestURI(url)
	req.Header.SetMethod("GET")

	// 设置默认头
	for k, v := range c.header {
		req.Header.Set(k, v)
	}

	// 设置请求头
	for k, v := range headers {
		req.Header.Set(k, v)
	}

	if err := c.client.Do(req, resp); err != nil {
		return nil, err
	}

	return resp, nil
}

// PostJSON 发送JSON POST请求
func (c *FastHttpClient) PostJSON(url string, data interface{}, headers map[string]string) (*fasthttp.Response, error) {
	req := fasthttp.AcquireRequest()
	defer fasthttp.ReleaseRequest(req)

	resp := fasthttp.AcquireResponse()
	defer fasthttp.ReleaseResponse(resp)

	req.SetRequestURI(url)
	req.Header.SetMethod("POST")
	req.Header.SetContentType("application/json")

	// 设置默认头
	for k, v := range c.header {
		req.Header.Set(k, v)
	}

	// 设置请求头
	for k, v := range headers {
		req.Header.Set(k, v)
	}

	// 序列化数据
	jsonData, err := json.Marshal(data)
	if err != nil {
		return nil, err
	}
	req.SetBody(jsonData)

	if err := c.client.Do(req, resp); err != nil {
		return nil, err
	}

	return resp, nil
}

// PostForm 发送表单POST请求
func (c *FastHttpClient) PostForm(url string, data map[string]string, headers map[string]string) (*fasthttp.Response, error) {
	req := fasthttp.AcquireRequest()
	defer fasthttp.ReleaseRequest(req)

	resp := fasthttp.AcquireResponse()
	defer fasthttp.ReleaseResponse(resp)

	req.SetRequestURI(url)
	req.Header.SetMethod("POST")
	req.Header.SetContentType("application/x-www-form-urlencoded")

	// 设置默认头
	for k, v := range c.header {
		req.Header.Set(k, v)
	}

	// 设置请求头
	for k, v := range headers {
		req.Header.Set(k, v)
	}

	// 设置表单数据
	args := fasthttp.AcquireArgs()
	defer fasthttp.ReleaseArgs(args)
	for k, v := range data {
		args.Add(k, v)
	}
	args.WriteTo(req.BodyWriter())

	if err := c.client.Do(req, resp); err != nil {
		return nil, err
	}

	return resp, nil
}

// Close 关闭客户端
func (c *FastHttpClient) Close() {
	// fasthttp.Client 不需要显式关闭
}

// OpenAPIKeyClient OpenAPI密钥认证客户端
type OpenAPIKeyClient struct {
	apiKey    string
	apiSecret string
	baseURL   string
	client    *FastHttpClient
}

// NewOpenAPIKeyClient 创建新的OpenAPI客户端
func NewOpenAPIKeyClient(baseURL, apiKey, apiSecret string) *OpenAPIKeyClient {
	// 确保baseURL以斜杠结尾
	if !strings.HasSuffix(baseURL, "/") {
		baseURL += "/"
	}

	option := FastHttpClientOption{
		Header: make(map[string]string),
	}
	client := NewFastHttpClient(option)

	return &OpenAPIKeyClient{
		apiKey:    apiKey,
		apiSecret: apiSecret,
		baseURL:   baseURL,
		client:    client,
	}
}

// generateSignature 生成HMAC-SHA256签名
func (c *OpenAPIKeyClient) generateSignature(text string) string {
	h := hmac.New(sha256.New, []byte(c.apiSecret))
	h.Write([]byte(text))
	return hex.EncodeToString(h.Sum(nil))
}

// buildAuthHeaders 构建认证请求头
func (c *OpenAPIKeyClient) buildAuthHeaders(requestBody string) map[string]string {
	headers := make(map[string]string)

	// 时间戳（毫秒）
	timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)

	// 随机数（Nonce）
	nonce := strings.ReplaceAll(uuid.New().String(), "-", "")

	// 签名计算：apiKey + timestamp + nonce + body
	signText := c.apiKey + timestamp + nonce + requestBody
	signature := c.generateSignature(signText)

	// 添加认证头
	headers["X-Api-Key"] = c.apiKey
	headers["X-Timestamp"] = timestamp
	headers["X-Nonce"] = nonce
	headers["X-Signature"] = signature
	headers["Content-Type"] = "application/json"

	return headers
}

// Get 发送GET请求（带签名认证）
func (c *OpenAPIKeyClient) Get(urlPath string, params map[string]string) (interface{}, error) {
	// 构建完整URL
	parsedBaseURL, err := url.Parse(c.baseURL)
	if err != nil {
		return nil, err
	}

	parsedPath, err := url.Parse(urlPath)
	if err != nil {
		return nil, err
	}

	fullURL := parsedBaseURL.ResolveReference(parsedPath).String()

	// 如果有参数，添加到URL
	if params != nil && len(params) > 0 {
		values := url.Values{}
		for k, v := range params {
			values.Add(k, v)
		}
		fullURL += "?" + values.Encode()
	}

	// GET请求的body为空字符串
	requestBody := ""
	authHeaders := c.buildAuthHeaders(requestBody)

	// 发送GET请求
	resp, err := c.client.Get(fullURL, authHeaders)
	if err != nil {
		return nil, err
	}

	// 检查响应状态
	statusCode := resp.StatusCode()
	if statusCode >= 400 {
		return nil, fmt.Errorf("HTTP %d: %s", statusCode, string(resp.Body()))
	}

	// 尝试解析JSON响应
	var result interface{}
	if err := json.Unmarshal(resp.Body(), &result); err != nil {
		return string(resp.Body()), nil
	}

	return result, nil
}

// Post 发送POST请求（带签名认证）
func (c *OpenAPIKeyClient) Post(urlPath string, data map[string]interface{}) (interface{}, error) {
	// 构建完整URL
	parsedBaseURL, err := url.Parse(c.baseURL)
	if err != nil {
		return nil, err
	}

	parsedPath, err := url.Parse(urlPath)
	if err != nil {
		return nil, err
	}

	fullURL := parsedBaseURL.ResolveReference(parsedPath).String()

	// 将数据转换为JSON字符串作为请求body
	requestBody, err := json.Marshal(data)
	if err != nil {
		return nil, err
	}

	authHeaders := c.buildAuthHeaders(string(requestBody))

	// 发送POST JSON请求
	resp, err := c.client.PostJSON(fullURL, data, authHeaders)
	if err != nil {
		return nil, err
	}

	// 检查响应状态
	statusCode := resp.StatusCode()
	if statusCode >= 400 {
		return nil, fmt.Errorf("HTTP %d: %s", statusCode, string(resp.Body()))
	}

	// 尝试解析JSON响应
	var result interface{}
	if err := json.Unmarshal(resp.Body(), &result); err != nil {
		return string(resp.Body()), nil
	}

	return result, nil
}

// PostForm 发送POST表单请求（带签名认证）
func (c *OpenAPIKeyClient) PostForm(urlPath string, data map[string]string) (interface{}, error) {
	// 构建完整URL
	parsedBaseURL, err := url.Parse(c.baseURL)
	if err != nil {
		return nil, err
	}

	parsedPath, err := url.Parse(urlPath)
	if err != nil {
		return nil, err
	}

	fullURL := parsedBaseURL.ResolveReference(parsedPath).String()

	// 将数据转换为JSON字符串作为请求body（签名需要）
	requestBody, err := json.Marshal(data)
	if err != nil {
		return nil, err
	}

	authHeaders := c.buildAuthHeaders(string(requestBody))

	// 发送POST表单请求
	resp, err := c.client.PostForm(fullURL, data, authHeaders)
	if err != nil {
		return nil, err
	}

	// 检查响应状态
	statusCode := resp.StatusCode()
	if statusCode >= 400 {
		return nil, fmt.Errorf("HTTP %d: %s", statusCode, string(resp.Body()))
	}

	// 尝试解析JSON响应
	var result interface{}
	if err := json.Unmarshal(resp.Body(), &result); err != nil {
		return string(resp.Body()), nil
	}

	return result, nil
}

// Close 关闭连接
func (c *OpenAPIKeyClient) Close() {
	if c.client != nil {
		c.client.Close()
	}
}

// OpenAPIKeyClientV2 OpenAPI密钥认证客户端（增强版）
type OpenAPIKeyClientV2 struct {
	apiKey    string
	apiSecret string
	baseURL   string
	client    *FastHttpClient
}

// NewOpenAPIKeyClientV2 创建新的增强版OpenAPI客户端
func NewOpenAPIKeyClientV2(baseURL, apiKey, apiSecret string, timeout *int, verifySSL bool, proxy *string) *OpenAPIKeyClientV2 {
	// 确保baseURL以斜杠结尾
	if !strings.HasSuffix(baseURL, "/") {
		baseURL += "/"
	}

	// 创建HTTP客户端配置
	headers := map[string]string{
		"User-Agent": "OpenAPI-Go-Client/1.0",
		"Accept":     "application/json",
	}

	socketTimeout := 0
	connectTimeout := 0
	if timeout != nil {
		socketTimeout = *timeout * 1000
		connectTimeout = *timeout * 1000
	}

	proxyAddress := ""
	if proxy != nil {
		proxyAddress = *proxy
	}

	option := FastHttpClientOption{
		Header:         headers,
		SocketTimeout:  socketTimeout,
		ConnectTimeout: connectTimeout,
		IgnoreSSL:      !verifySSL,
		ProxyAddress:   proxyAddress,
	}

	client := NewFastHttpClient(option)

	return &OpenAPIKeyClientV2{
		apiKey:    apiKey,
		apiSecret: apiSecret,
		baseURL:   baseURL,
		client:    client,
	}
}

// generateSignatureV2 生成增强版签名
func (c *OpenAPIKeyClientV2) generateSignatureV2(method, path, timestamp, nonce, body string) string {
	// 计算body哈希
	bodyHash := ""
	if body != "" {
		h := sha256.New()
		h.Write([]byte(body))
		bodyHash = hex.EncodeToString(h.Sum(nil))
	}

	// 构建签名字符串
	signText := fmt.Sprintf("%s%s%s%s%s%s", strings.ToUpper(method), path, c.apiKey, timestamp, nonce, bodyHash)

	// 生成HMAC-SHA256签名
	h := hmac.New(sha256.New, []byte(c.apiSecret))
	h.Write([]byte(signText))
	return hex.EncodeToString(h.Sum(nil))
}

// buildAuthHeadersV2 构建增强版认证请求头
func (c *OpenAPIKeyClientV2) buildAuthHeadersV2(method, path, body string) map[string]string {
	headers := make(map[string]string)

	// 时间戳（秒，整数）
	timestamp := strconv.FormatInt(time.Now().Unix(), 10)

	// 随机数（Nonce）
	nonce := strings.ReplaceAll(uuid.New().String(), "-", "")

	// 生成签名
	signature := c.generateSignatureV2(method, path, timestamp, nonce, body)

	// 添加认证头
	headers["X-Api-Key"] = c.apiKey
	headers["X-Timestamp"] = timestamp
	headers["X-Nonce"] = nonce
	headers["X-Signature"] = signature
	headers["Content-Type"] = "application/json"

	return headers
}

// Request 发送通用请求（带签名认证）
func (c *OpenAPIKeyClientV2) Request(method, urlPath string, data map[string]interface{}, params map[string]string) (interface{}, error) {
	// 构建完整URL
	parsedBaseURL, err := url.Parse(c.baseURL)
	if err != nil {
		return nil, err
	}

	parsedPath, err := url.Parse(urlPath)
	if err != nil {
		return nil, err
	}

	fullURL := parsedBaseURL.ResolveReference(parsedPath).String()

	// 如果有参数，添加到URL
	if params != nil && len(params) > 0 {
		values := url.Values{}
		for k, v := range params {
			values.Add(k, v)
		}
		fullURL += "?" + values.Encode()
	}

	// 解析URL以获取路径部分（用于签名）
	parsedURL, err := url.Parse(fullURL)
	if err != nil {
		return nil, err
	}
	path := parsedURL.Path
	if parsedURL.RawQuery != "" {
		path += "?" + parsedURL.RawQuery
	}

	// 准备请求体
	requestBody := ""
	if data != nil {
		jsonData, err := json.Marshal(data)
		if err != nil {
			return nil, err
		}
		requestBody = string(jsonData)
	}

	// 构建认证头
	authHeaders := c.buildAuthHeadersV2(method, path, requestBody)

	var resp *fasthttp.Response

	// 根据方法发送请求
	switch strings.ToUpper(method) {
	case "GET":
		resp, err = c.client.Get(fullURL, authHeaders)
	case "POST":
		resp, err = c.client.PostJSON(fullURL, data, authHeaders)
	case "PUT":
		// 注意：这里简化为POST，实际应用中应实现PUT方法
		resp, err = c.client.PostJSON(fullURL, data, authHeaders)
	case "DELETE":
		// 注意：这里简化为GET，实际应用中应实现DELETE方法
		resp, err = c.client.Get(fullURL, authHeaders)
	default:
		return nil, fmt.Errorf("不支持的HTTP方法: %s", method)
	}

	if err != nil {
		return nil, err
	}

	// 检查响应状态
	statusCode := resp.StatusCode()
	if statusCode >= 400 {
		return nil, fmt.Errorf("HTTP %d: %s", statusCode, string(resp.Body()))
	}

	// 尝试解析JSON响应
	var result interface{}
	if err := json.Unmarshal(resp.Body(), &result); err != nil {
		return string(resp.Body()), nil
	}

	return result, nil
}

// Close 关闭连接
func (c *OpenAPIKeyClientV2) Close() {
	if c.client != nil {
		c.client.Close()
	}
}

File: python/src/OpenAPIKeyClient.py
# OpenAPIKeyClient.py
import json
import hashlib
import hmac
import time
import uuid
from typing import Dict, Any, Optional
from urllib.parse import urljoin

# 导入之前翻译的 HttpClient
from HttpClient import HttpClient, HttpClientOption


class OpenAPIKeyClient:
    """OpenAPI 密钥认证客户端"""

    def __init__(self, base_url: str, api_key: str, api_secret: str):
        """
        初始化 OpenAPI 客户端

        Args:
            base_url: API 基础URL
            api_key: API 密钥
            api_secret: API 密钥
        """
        self.api_key = api_key
        self.api_secret = api_secret
        self.base_url = base_url.rstrip('/') + '/'  # 确保以斜杠结尾

        # 创建 HTTP 客户端
        option = HttpClientOption(header={})
        self.client = HttpClient(option)

    def _generate_signature(self, text: str) -> str:
        """
        生成 HMAC-SHA256 签名

        Args:
            text: 待签名的文本

        Returns:
            十六进制签名字符串
        """
        try:
            # 创建 HMAC-SHA256 签名
            signature = hmac.new(
                self.api_secret.encode('utf-8'),
                text.encode('utf-8'),
                hashlib.sha256
            )

            # 返回十六进制字符串
            return signature.hexdigest()
        except Exception as e:
            raise RuntimeError(f"签名生成失败: {str(e)}")

    def _build_auth_headers(self, request_body: str = "") -> Dict[str, str]:
        """
        构建认证请求头

        Args:
            request_body: 请求体内容

        Returns:
            认证头字典
        """
        headers = {}

        # 时间戳（毫秒）
        timestamp = str(int(time.time() * 1000))

        # 随机数（Nonce）
        nonce = str(uuid.uuid4()).replace("-", "")

        # 签名计算：apiKey + timestamp + nonce + body
        sign_text = self.api_key + timestamp + nonce + request_body
        signature = self._generate_signature(sign_text)

        # 添加认证头
        headers["X-Api-Key"] = self.api_key
        headers["X-Timestamp"] = timestamp
        headers["X-Nonce"] = nonce
        headers["X-Signature"] = signature
        headers["Content-Type"] = "application/json"

        return headers

    def get(self, url: str, params: Optional[Dict[str, Any]] = None) -> Any:
        """
        发送 GET 请求（带签名认证）

        Args:
            url: 相对URL路径
            params: 查询参数

        Returns:
            响应数据

        Raises:
            Exception: 请求失败时抛出
        """
        # 构建完整URL
        full_url = urljoin(self.base_url, url.lstrip('/'))

        # GET 请求的 body 为空字符串
        request_body = ""
        auth_headers = self._build_auth_headers(request_body)

        # 发送 GET 请求
        response = self.client.get(full_url, headers=auth_headers)

        # 检查响应状态
        if response.status_code >= 400:
            raise Exception(f"HTTP {response.status_code}: {response.text}")

        # 尝试解析 JSON 响应
        try:
            return response.json()
        except json.JSONDecodeError:
            return response.text

    def post(self, url: str, data: Dict[str, Any]) -> Any:
        """
        发送 POST 请求（带签名认证）

        Args:
            url: 相对URL路径
            data: 请求数据

        Returns:
            响应数据

        Raises:
            Exception: 请求失败时抛出
        """
        # 构建完整URL
        full_url = urljoin(self.base_url, url.lstrip('/'))

        # 将数据转换为 JSON 字符串作为请求 body
        request_body = json.dumps(data, ensure_ascii=False)
        auth_headers = self._build_auth_headers(request_body)

        # 发送 POST JSON 请求
        response = self.client.post_json(full_url, data, headers=auth_headers)

        # 检查响应状态
        if response.status_code >= 400:
            raise Exception(f"HTTP {response.status_code}: {response.text}")

        # 尝试解析 JSON 响应
        try:
            return response.json()
        except json.JSONDecodeError:
            return response.text

    def post_form(self, url: str, data: Dict[str, Any]) -> Any:
        """
        发送 POST 表单请求（带签名认证）

        Args:
            url: 相对URL路径
            data: 表单数据

        Returns:
            响应数据

        Raises:
            Exception: 请求失败时抛出
        """
        # 构建完整URL
        full_url = urljoin(self.base_url, url.lstrip('/'))

        # 将数据转换为 JSON 字符串作为请求 body（签名需要）
        request_body = json.dumps(data, ensure_ascii=False)
        auth_headers = self._build_auth_headers(request_body)

        # 发送 POST 表单请求
        response = self.client.post_form(full_url, data, headers=auth_headers)

        # 检查响应状态
        if response.status_code >= 400:
            raise Exception(f"HTTP {response.status_code}: {response.text}")

        # 尝试解析 JSON 响应
        try:
            return response.json()
        except json.JSONDecodeError:
            return response.text

    def close(self) -> None:
        """关闭连接"""
        if self.client:
            self.client.close()

    def __enter__(self):
        """上下文管理器入口"""
        return self

    def __exit__(self, exc_type, exc_val, exc_tb):
        """上下文管理器退出"""
        self.close()


class OpenAPIKeyClientV2:
    """
    OpenAPI 密钥认证客户端（增强版）
    支持更多认证方式和配置选项
    """

    def __init__(self,
                 base_url: str,
                 api_key: str,
                 api_secret: str,
                 timeout: Optional[int] = 30,
                 verify_ssl: bool = True,
                 proxy: Optional[str] = None):
        """
        初始化 OpenAPI 客户端（增强版）

        Args:
            base_url: API 基础URL
            api_key: API 密钥
            api_secret: API 密钥
            timeout: 超时时间（秒）
            verify_ssl: 是否验证 SSL 证书
            proxy: 代理地址（格式: "host:port"）
        """
        self.api_key = api_key
        self.api_secret = api_secret
        self.base_url = base_url.rstrip('/') + '/'

        # 创建 HTTP 客户端配置
        headers = {
            "User-Agent": "OpenAPI-Python-Client/1.0",
            "Accept": "application/json"
        }

        option = HttpClientOption(
            header=headers,
            socket_timeout=timeout * 1000 if timeout else None,
            connect_timeout=timeout * 1000 if timeout else None,
            ignore_ssl=not verify_ssl,
            proxy_address=proxy
        )

        self.client = HttpClient(option)

    def _generate_signature_v2(self,
                               method: str,
                               path: str,
                               timestamp: str,
                               nonce: str,
                               body: str = "") -> str:
        """
        生成增强版签名

        Args:
            method: HTTP 方法（GET, POST等）
            path: 请求路径（不包含域名）
            timestamp: 时间戳
            nonce: 随机数
            body: 请求体

        Returns:
            十六进制签名字符串
        """
        # 签名算法：method + path + apiKey + timestamp + nonce + bodyHash
        body_hash = hashlib.sha256(body.encode('utf-8')).hexdigest() if body else ""

        # 构建签名字符串
        sign_text = f"{method.upper()}{path}{self.api_key}{timestamp}{nonce}{body_hash}"

        # 生成 HMAC-SHA256 签名
        signature = hmac.new(
            self.api_secret.encode('utf-8'),
            sign_text.encode('utf-8'),
            hashlib.sha256
        ).hexdigest()

        return signature

    def _build_auth_headers_v2(self,
                               method: str,
                               path: str,
                               body: str = "") -> Dict[str, str]:
        """
        构建增强版认证请求头

        Args:
            method: HTTP 方法
            path: 请求路径
            body: 请求体内容

        Returns:
            认证头字典
        """
        headers = {}

        # 时间戳（秒，整数）
        timestamp = str(int(time.time()))

        # 随机数（Nonce）
        nonce = str(uuid.uuid4()).replace("-", "")

        # 生成签名
        signature = self._generate_signature_v2(method, path, timestamp, nonce, body)

        # 添加认证头
        headers["X-Api-Key"] = self.api_key
        headers["X-Timestamp"] = timestamp
        headers["X-Nonce"] = nonce
        headers["X-Signature"] = signature
        headers["Content-Type"] = "application/json"

        return headers

    def request(self,
                method: str,
                url: str,
                data: Optional[Dict[str, Any]] = None,
                params: Optional[Dict[str, Any]] = None) -> Any:
        """
        发送通用请求（带签名认证）

        Args:
            method: HTTP 方法（GET, POST, PUT, DELETE等）
            url: 相对URL路径
            data: 请求体数据
            params: 查询参数

        Returns:
            响应数据

        Raises:
            Exception: 请求失败时抛出
        """
        # 构建完整URL
        full_url = urljoin(self.base_url, url.lstrip('/'))

        # 提取路径部分（用于签名）
        parsed_url = urlparse(full_url)
        path = parsed_url.path
        if parsed_url.query:
            path += "?" + parsed_url.query

        # 准备请求体
        request_body = ""
        if data is not None:
            request_body = json.dumps(data, ensure_ascii=False)

        # 构建认证头
        auth_headers = self._build_auth_headers_v2(method, path, request_body)

        # 根据方法发送请求
        if method.upper() == "GET":
            response = self.client.get(full_url, headers=auth_headers)
        elif method.upper() == "POST":
            response = self.client.post_json(full_url, data, headers=auth_headers)
        elif method.upper() == "PUT":
            # 注意：HttpClient 类没有实现 PUT 方法，这里简化为 POST
            # 在实际应用中，可能需要扩展 HttpClient 类
            response = self.client.post_json(full_url, data, headers=auth_headers)
        elif method.upper() == "DELETE":
            # 注意：HttpClient 类没有实现 DELETE 方法
            # 在实际应用中，可能需要扩展 HttpClient 类
            response = self.client.get(full_url, headers=auth_headers)
        else:
            raise ValueError(f"不支持的 HTTP 方法: {method}")

        # 检查响应状态
        if response.status_code >= 400:
            raise Exception(f"HTTP {response.status_code}: {response.text}")

        # 尝试解析 JSON 响应
        try:
            return response.json()
        except json.JSONDecodeError:
            return response.text

    def close(self) -> None:
        """关闭连接"""
        if self.client:
            self.client.close()

    def __enter__(self):
        """上下文管理器入口"""
        return self

    def __exit__(self, exc_type, exc_val, exc_tb):
        """上下文管理器退出"""
        self.close()


